components/net/http_cache.rs

```diff
@@ -70,6 +72,22 @@ struct CachedResource {
     awaiting_body: Arc<Mutex<Vec<Sender<Data>>>>
 }
 
+impl MallocSizeOf for CachedResource {
```

Comment:
What if we create a `struct Measurable<T: MallocSizeOf>(pub T)` that derefs to the inner value in the malloc_size_of crate? Then we could do this:

```rust
#[derive(Clone, MallocSizeOf)]
struct MeasurableCachedResource {
     metadata: CachedMetadata,
     location_url: Option<Result<ServoUrl, String>>,
     https_state: HttpsState,
     status: Option<StatusCode>,
     raw_status: Option<(u16, Vec<u8>)>,
     url_list: Vec<ServoUrl>,
     expires: Duration,
     last_validated: Tm,
}

 struct CachedResource {
     request_headers: Arc<Mutex<Headers>>,
     body: Arc<Mutex<ResponseBody>>,
     aborted: Arc<AtomicBool>,
     awaiting_body: Arc<Mutex<Vec<Sender<Data>>>>,
     data: Measurable<MeasurableCachedResource>,
 }
```
This would make it easier to verify the manual implementation of size_of for CachedResource.

Reply:
So to access the measurable resource fields, the result of this would be something like, `cachedresource.data.0.https_state` or `(*cachedresource.data).https_state` if Deref is implemented for Measurable?

Reply:
Yeah, `cachedresource.data.https_state` should be enough. If anything needs to update the fields, you can implement DerefMut for Measurable as well.